src/main/java/org/example/servletsHomework/servlet/TextsServlet.java

```diff
@@ -0,0 +1,153 @@
+                textService.addText(userId, new Texts(textId, text));
+
+                responseBody.setTextId(textId);
+                responseBody.setText(text);
+                responseBody.setMessage("Text saved with id: " + textId);
+
+                resp.setStatus(HttpServletResponse.SC_CREATED);
+                respWriter.write(objectMapper.writeValueAsString(responseBody));
+            }
+        } catch (IOException e) {
+            throw new InternalServerErrorException(INTERNAL_SERVER_ERROR);
+        }
+
+    }
+
+
+    @Override
+    public void handleGet(HttpServletRequest req, HttpServletResponse resp) {
+
+        String token = req.getHeader(HEADER_NAME_TOKEN);
+        long userId = tokensAndUserStorage.getUserIdByToken(token);
+
+        try {
+            PrintWriter respWriter = resp.getWriter();
+            String pathInfo = req.getPathInfo();
+            if (pathInfo.equals("/")) {
+                respWriter.write(textService.getAllTexts(userId).toString());
+                resp.setStatus(HttpServletResponse.SC_OK);
+            } else {
+                String[] pathParts = pathInfo.split("/");
+                if (pathParts.length == 2) {
+                    respWriter.write(handleTextRequest(userId, pathParts[1], "get", pathInfo));
+                    resp.setStatus(HttpServletResponse.SC_OK);
+                } else {
+                    throw new BadRequestException(INVALID_REQUEST_PATH + ": " + pathInfo);
+                }
+            }
+        } catch (IOException ex) {
+            throw new InternalServerErrorException(INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    @Override
+    public void handleDelete(HttpServletRequest req, HttpServletResponse resp) {
+
+        String token = req.getHeader(HEADER_NAME_TOKEN);
+        long userId = tokensAndUserStorage.getUserIdByToken(token);
+
+        try {
+            PrintWriter respWriter = resp.getWriter();
+            String pathInfo = req.getPathInfo();
+            if (pathInfo.equals("/deleteAll")) {
+                textService.deleteAll(userId);
+                resp.setStatus(HttpServletResponse.SC_OK);
+                respWriter.write("All text has been deleted");
+            } else {
+                String[] pathParts = pathInfo.split("/");
+                if (pathParts.length == 2) {
+                    respWriter.write(handleTextRequest(userId, pathParts[1], "delete", pathInfo));
+                    resp.setStatus(HttpServletResponse.SC_OK);
+                    textService.deleteTextById(userId, Long.parseLong(pathParts[1]));
+                } else {
+                    throw new BadRequestException(INVALID_REQUEST_PATH + ": " + pathInfo);
+                }
+            }
+        } catch (IOException ex) {
+            throw new InternalServerErrorException(INTERNAL_SERVER_ERROR);
+        }
+    }
+
+
+    private String handleTextRequest(Long idUser, String idText, String action, String pathInfo) {
```

Comment:
why everywhere you call it `userId` or `textId` but here decided to break this logic and named it `idUser` and `idText` ?

Reply:
Can you explain what this method doing in human words ? Not just reading code. Also try to match this explanation with SOLID principles. What is response in the method. Just write simple java doc to not make another developer dive into code of the method